Parsers/ASimRegistryEvent/Parsers/vimRegistryEventSentinelOne.yaml

```diff
@@ -0,0 +1,131 @@
+ParserName: vimRegistryEventSentinelOne
+EquivalentBuiltInParser: _Im_RegistryEvent_SentinelOne
+ParserParams:
+  - Name: disabled
+    Type: bool
+    Default: false
+ParserQuery: |
+  let EventTypeLookup = datatable (alertInfo_eventType_s: string, EventType: string)
+  [
+      "REGVALUEMODIFIED", "RegistryValueSet",
+      "REGVALUECREATE", "RegistryValueSet",
+      "REGKEYCREATE", "RegistryKeyCreated",
+      "REGKEYDELETE", "RegistryKeyDeleted",
+      "REGVALUEDELETE", "RegistryValueDeleted",
+      "REGKEYRENAME", "RegistryKeyRenamed"
+  ];
+  let RegistryKeyPrefixLookup = datatable (
+      RegistryKeyPrefix: string,
+      RegistryKeyNormalizedPrefix: string
+  )
+  [
+      "MACHINE", "HKEY_LOCAL_MACHINE",
+      "USER", "HKEY_USERS",
+      "CONFIG", "HKEY_CURRENT_CONFIG",
+      "ROOT", "HKEY_CLASSES_ROOT"
+  ];
+  let RegistryPreviousValueTypeLookup = datatable (
+      alertInfo_registryOldValueType_s: string,
+      RegistryPreviousValueType_lookup: string
+  )
+  [
+      "BINARY", "Reg_Binary",
+      "DWORD", "Reg_DWord",
+      "QWORD", "Reg_QWord",
+      "SZ", "Reg_Sz",
+      "EXPAND_SZ", "Reg_Expand_Sz",
+      "MULTI_SZ", "Reg_Multi_Sz",
+      "DWORD_BIG_ENDIAN", "Reg_DWord"
+  ];
+  let parser = (disabled: bool=false) { 
+      SentinelOne_CL
+      | where not(disabled)
+          and event_name_s == "Alerts."
+          and alertInfo_eventType_s in ("REGVALUEMODIFIED", "REGVALUECREATE", "REGKEYCREATE", "REGKEYDELETE", "REGVALUEDELETE", "REGKEYRENAME")
+      | lookup EventTypeLookup on alertInfo_eventType_s
+      | lookup RegistryPreviousValueTypeLookup on alertInfo_registryOldValueType_s
+      | invoke _ASIM_ResolveDvcFQDN('agentDetectionInfo_name_s')
+      | extend RegistryKeyPrefix = tostring(split(alertInfo_registryKeyPath_s, @'\')[0])
+      | lookup RegistryKeyPrefixLookup on RegistryKeyPrefix
+      | extend RegistryKey = replace_string(alertInfo_registryKeyPath_s, RegistryKeyPrefix, RegistryKeyNormalizedPrefix)
+      | extend RegistryValue = iff(alertInfo_eventType_s in ("REGVALUEMODIFIED", "REGVALUECREATE", "REGVALUEDELETE") , tostring(split(alertInfo_registryKeyPath_s, @'\')[-1]), "")
+      | extend RegistryValueType = case(
+                                 alertInfo_registryValue_s matches regex '^[0-9]+$', "Reg_Dword",
+                                 alertInfo_registryValue_s startswith "0x" and strlen(alertInfo_registryValue_s) <= 10, "Reg_DWord",
+                                 alertInfo_registryValue_s startswith "0x" and strlen(alertInfo_registryValue_s) > 10, "Reg_QWord",
+                                 alertInfo_registryValue_s matches regex '^[A-Fa-f0-9]+$', "Reg_Binary",
```

Comment:
Can we avoid regex?

Reply:
Registry Value type is not directly available in any specific field. There might be chance to get Registry Value data from "alertInfo_registryValue_s" field that is why we used regex to derive type based on value.